### PR TITLE
adding the commit for slice containing None

### PR DESCRIPTION
**Pull request by abit2**
this makes operations such as
```
sequence.index_as_cube[:] possible
```
before it did not support sequence.index_as_cube[:]

### CODE REVIEW

**Comment by abit2** [comment]
@DanRyanIrish maybe I should add the dict(self.meta) in IRISPY

File: sunpycube/cube/cube_utils.py [hunk 1]
```diff
@@ -665,11 +665,21 @@ def _convert_cube_like_index_to_sequence_indices(cube_like_index, cumul_cube_len
...
 
 def _convert_cube_like_slice_to_sequence_slices(cube_like_slice, cumul_cube_lengths):
+
+    start = cube_like_slice.start if cube_like_slice.start is not None else 0
+    stop = cube_like_slice.stop if cube_like_slice.stop is not None else cumul_cube_lengths[-1]
+    if stop > cumul_cube_lengths[-1]:
+        stop = cumul_cube_lengths[-1]
+    if start < 0:
+        start = 0
+    step = cube_like_slice.step if cube_like_slice.step is not None else 1
```
Comment (DanRyanIrish) [inline sunpycube/cube/cube_utils.py]: Does this have consequences of negative indexing? e.g. ```CubeSequence.index_as_cube[-1]```

**Comment by abit2** [comment]
made changes to the tests in the test_cube_utils.py

**Comment by abit2** [comment]
@DanRyanIrish I think it is good to go :)

**Review by DanRyanIrish — APPROVED** [review]

**Pull request merged**